benches/parser.rs

```diff
@@ -0,0 +1,60 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use std::{fs, path::PathBuf, str::FromStr};
+
+fn benchmark_quil_corpus(c: &mut Criterion) {
+    from_corpus().iter().for_each(|cfg| {
+        c.bench_function(&cfg.name, |b| {
+            b.iter(|| {
+                let _ = quil_rs::Program::from_str(&cfg.program);
+            })
+        });
+    })
+}
+
+struct QuilBenchConfig {
+    name: String,
+    program: String,
+}
+
+fn from_corpus() -> Vec<QuilBenchConfig> {
+    const PATH_SRC: &str = "benches/quilc/tests/good-test-files";
+
+    // collect valid quil programs
+    let mut programs = vec![];
+    let mut corpus_dir = PathBuf::new();
+    PATH_SRC.split('/').for_each(|p| corpus_dir.push(p));
+    let dir = fs::read_dir(corpus_dir).expect("failed to locate quil corpus directory");
```

Comment:
I think you can do this:

```suggestion
    const PATH_SRC: &str = "benches/quilc/tests/good-test-files";

    // collect valid quil programs
    let mut programs = vec![];
    let mut corpus_dir = PathBuf::from(PATH_SRC);
    PATH_SRC.split('/').for_each(|p| corpus_dir.push(p));
    let dir = fs::read_dir(corpus_dir).expect("failed to locate quil corpus directory");
```

Or, better yet, this:

```suggestion
    const PATH_SRC: &str = "benches/quilc/tests/good-test-files";

    // collect valid quil programs
    let mut programs = vec![];
    let dir = fs::read_dir(PATH_SRC).expect("failed to locate quil corpus directory");
```

Reply:
d'oh good catch! going with `PathBuf::from`, not sure if `read_dir` is platform independent.

Reply:
*update*: I reverted the change to use `PathBuf::from` here since I am pretty sure both are dependent on the running platform.